openBCIBoard.js

```diff
@@ -75,60 +94,78 @@ function OpenBCIFactory() {
     util.inherits(OpenBCIBoard, stream.Stream);
 
     /**
-     * Purpose: The essential precursor method to be called initially to establish a
+     * @description The essential precursor method to be called initially to establish a
      *              serial connection to the OpenBCI board.
      * @param portName - a string that contains the port name of the OpenBCIBoard.
      * @returns {Promise} if the board was able to connect. If at any time the serial port
      *              closes or errors then this promise will be rejected, and this should be
      *              observed and taken care of in the most front facing user methods.
-     * Author: AJ Keller (@pushtheworldllc)
+     * @author AJ Keller (@pushtheworldllc)
      */
     OpenBCIBoard.prototype.connect = function(portName) {
         this.connected = false;
 
         return new Promise((resolve,reject) => {
-            var boardSerial = new serialPort.SerialPort(portName, {
-                baudRate: this.options.baudRate
-            },(err) => {
-                reject(err);
-            });
+            // If we are simulating, set boardSerial to fake name
+            var boardSerial;
+            /* istanbul ignore else */
+            if (this.options.simulate) {
+                if (this.options.verbose) console.log('using faux board ' + portName);
+                boardSerial = new openBCISimulator.OpenBCISimulator(portName, {
+                    verbose: this.options.verbose
+                });
+            } else {
+                /* istanbul ignore if */
+                if (this.options.verbose) console.log('using real board ' + portName);
```

Comment:
You can start it a number of ways... * Sending connect "/dev/tty.openBCISimulator"* Setting `simulate` true on constructor call* calling `simulatorStart()`

AJ Keller
Founder 
Push The World
843-471-0039




On Fri, Feb 5, 2016 at 9:58 AM -0800, "Teon L Brooks" <notifications@github.com> wrote:












In openBCIBoard.js:
> -            var boardSerial = new serialPort.SerialPort(portName, {
> -                baudRate: this.options.baudRate
> -            },(err) => {
> -                reject(err);
> -            });
> +            // If we are simulating, set boardSerial to fake name
> +            var boardSerial;
> +            /* istanbul ignore else */
> +            if (this.options.simulate) {
> +                if (this.options.verbose) console.log('using faux board ' + portName);
> +                boardSerial = new openBCISimulator.OpenBCISimulator(portName, {
> +                    verbose: this.options.verbose
> +                });
> +            } else {
> +                /* istanbul ignore if */
> +                if (this.options.verbose) console.log('using real board ' + portName);



so the simulator requires a portname?



—
Reply to this email directly or view it on GitHub.